### PR TITLE
Security audit logging

### PR DESCRIPTION
**Pull request by cheungpat**
connects #150

### CODE REVIEW

File: features/150-audit-log/api.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Security audit loggin on Skygear Cloud
```
Comment (cheungpat) [inline features/150-audit-log/api.md]: s/loggin/login/

**Comment by cheungpat** [comment]
* how long the s3 retain log
* how often the s3 will update the log

**Comment by chpapa** [comment]
> how long the s3 retain log

I guess this one need to be configurable.

**Comment by cheungpat** [comment]
> I guess this one need to be configurable.

yea, i guess this should be configurable, but since the log is not retained by the skygear server, the configuration will apply to the cloud daemons instead. To make things easier can we say the cluster will have the same audit log retention period that is long enough for most apps?

**Comment by chpapa** [comment]
> To make things easier can we say the cluster will have the same audit log retention period that is long enough for most apps?

My gut feeling is yes.

File: features/150-audit-log/api.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+The portal is expected to call this API to obtain a list of audit-logs for
+display.
+
+The controller is expected to check that the user is an owner or collaborator
+of an app. Then the controller lists all objects with a certain prefix
+from the S3 bucket that is storing the audit logs.
+
+* S3 bucket: TBD
+* Prefix: `audit-logs/<app>/<date>/`
```
Comment (rickmak) [inline features/150-audit-log/api.md]: Will the format change to this is better for query `<month>/<week>/<day>`

File: features/150-audit-log/api.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+log events might not be immediately available.
+
+## Controller API
+
+The Controller API does not need to update to enable audit logging. Portal
+will add environment variable to config.
+
+The Controller will add the following APIs:
+
+### `GET /apps/<id>/audit-logs/<date>`
```
Comment (rickmak) [inline features/150-audit-log/api.md]: Skycli can provide command to download a date range.

**Pull request merged**